protostar-rust/src/lib.rs

```diff
@@ -1,84 +1,126 @@
-use anyhow::{anyhow, Context, Result};
-use cairo_lang_protostar::test_collector::{collect_tests, LinkedLibrary};
-use cairo_lang_runner::{RunResultValue, SierraCasmRunner};
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use anyhow::{Context, Result};
 use camino::{Utf8Path, Utf8PathBuf};
 use serde::Deserialize;
 use walkdir::WalkDir;
 
+use cairo_lang_protostar::casm_generator::TestConfig;
+use cairo_lang_protostar::test_collector::{collect_tests, LinkedLibrary};
+use cairo_lang_runner::{SierraCasmRunner, StarknetState};
+use cairo_lang_sierra::program::Program;
+use cairo_lang_sierra_to_casm::metadata::MetadataComputationConfig;
+
+use crate::test_stats::TestsStats;
+
+pub mod pretty_printing;
+mod test_stats;
+
 #[derive(Deserialize, Debug)]
 pub struct ProtostarTestConfig {
     #[serde(default)]
     exit_first: bool, // TODO Not implemented!
 }
 
-fn run_result_value_to_string(run_result: RunResultValue) -> String {
-    return match run_result {
-        RunResultValue::Success(data) => format!("PASS {:?}", data),
-        RunResultValue::Panic(data) => format!("FAIL {:?}", data),
-    };
-}
+type TestInstance = (Program, Vec<TestConfig>, Utf8PathBuf);
```

Comment:
A type alias like this isn't really useful for other developers, I'd argue it even makes everything less readable. 

Let's create an proper struct for this.